kpops/components/common/kubernetes_model.py

```diff
@@ -1,12 +1,85 @@
 import enum
+from collections.abc import Iterator
+from typing import Any
 
-from pydantic import Field
+import yaml
+from pydantic import ConfigDict, Field
+from typing_extensions import override
 
 from kpops.utils.docstring import describe_attr
-from kpops.utils.pydantic import DescConfigModel
+from kpops.utils.pydantic import CamelCaseConfigModel, DescConfigModel
 
 # Matches plain integer or numbers with valid suffixes: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory
 MEMORY_PATTERN = r"^\d+([EPTGMk]|Ei|Pi|Ti|Gi|Mi|Ki)?$"
+K8S_LABEL_MAX_LEN = 63
+
+
+class ManagedFieldsEntry(CamelCaseConfigModel):
+    # Define this class based on its actual structure
+    fields: dict[str, Any] | None = None
+
+
+class OwnerReference(CamelCaseConfigModel):
+    # Define this class based on its actual structure
+    apiVersion: str
+    kind: str
+    name: str
+    uid: str
+    controller: bool | None = None
+    block_owner_deletion: bool | None = None
+
+
+class ObjectMeta(CamelCaseConfigModel):
+    """Metadata for all Kubernetes objects."""
+
+    annotations: dict[str, str] | None = None
+    creation_timestamp: str | None = Field(
+        default=None, description="Timestamp in RFC3339 format"
+    )
+    deletion_grace_period_seconds: int | None = None
+    deletion_timestamp: str | None = Field(
+        default=None, description="Timestamp in RFC3339 format"
+    )
+    finalizers: list[str] | None = None
+    generate_name: str | None = None
+    generation: int | None = None
+    labels: dict[str, str] | None = None
+    managed_fields: list[ManagedFieldsEntry] | None = None
+    name: str | None = None
+    namespace: str | None = None
+    owner_references: list[OwnerReference] | None = None
+    resource_version: str | None = None
+    self_link: str | None = Field(
+        default=None,
+        description="Deprecated field, not populated by Kubernetes in modern versions",
+    )
+    uid: str | None = None
```

Comment:
do we need all of those? I feel like many have no meaning to KPOps

Reply:
They are a representation of the lightkube ObjectMeta. They are documented here:

https://gtsystem.github.io/lightkube-models/1.19/models/meta_v1/#objectmeta

We can have them removed and allow extra fields.

Reply:
yeah, I think I would prefer a reduction to those fields that are most likely to be needed in KPOps